openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -136,15 +136,18 @@ def award_program_certificates(self, username):
     """
     LOGGER.info('Running task award_program_certificates for username %s', username)
 
+    config = ProgramsApiConfig.current()
+    countdown = 2 ** self.request.retries
+
     # If either programs or credentials config models are disabled for this
     # feature, this task should not have been invoked in the first place, and
     # an error somewhere is likely (though a race condition is also possible).
     # In either case, the task should not be executed nor should it be retried.
-    if not ProgramsApiConfig.current().is_certification_enabled:
+    if not config.is_certification_enabled:
         LOGGER.warning(
             'Task award_program_certificates cannot be executed when program certification is disabled in API config',
         )
-        return
+        raise self.retry(countdown=countdown, max_retries=config.max_retries)
```

Comment:
I think it might also be appropriate to retry when credentials config is disabled. At the moment, the task is skipped when display of certificates is disabled.